File: super-heroes-frontend/src/app/models/heroi.ts
import { SuperPoder } from './super-poder'; // ajuste o caminho se necessário

export interface Heroi {
  id: number;
  nome: string;
  nomeHeroi: string;
  superpoderes: SuperPoder[]; // agora é uma lista de objetos
  dataNascimento: Date | null;
  altura: number;
  peso: number;
}

File: SuperHeroes/Data/Heroi.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SuperHeroes.Data
{
    public class Heroi
    {
        [Key]
        public int Id { get; set; }

        [Column(TypeName = "varchar(120)")]
        public required string Nome { get; set; } 
        
        [Column(TypeName = "varchar(120)")]
        public required string NomeHeroi { get; set; }
        
        [Column(TypeName = "date")]
        public required DateTime DataNascimento { get; set; }

        public required float Altura { get; set; }

        public required float Peso { get; set; }

        public ICollection<SuperPoder> Superpoderes { get; set; } = null!;
    }
}
